src/glmfit.jl

```diff
@@ -233,10 +233,13 @@ function _fit!(m::AbstractGLM, verbose::Bool, maxIter::Integer, minStepFac::Real
         installbeta!(p, f)
         crit = (devold - dev)/dev
         verbose && println("$i: $dev, $crit")
-        if crit < convTol
+        if crit < convTol || dev == 0
             cvg = true
             break
         end
+        if !isfinite(crit)
+            error("convergence criterion is $crit") # maybe change ConvergenceException to allow for custom error
+        end
```

Comment:
I think I've changed my mind on this particular error. I don't think it should ever happen that we get `NaN`s or `Inf`s here so this should probably just be an `@assert`. The `ConvergeException` should be thrown for cases that could possibly happen for certain datasets.